src/source/promises.bs

```diff
@@ -251,7 +251,13 @@ namespace promises.internal
         if storage = invalid then
             ' unregister any observers on the promise to prevent multiple callbacks
             promises.internal.unobserveFieldScoped(promise, promises.internal.PromiseField.promiseState)
-            promises.internal.observeFieldScoped(promise, promises.internal.PromiseField.promiseState, promises.internal.notifyListeners)
+            promises.internal.observeFieldScoped(promise, promises.internal.PromiseField.promiseState, sub(event)
```

Comment:
This was the important piece here. This observer fires _synchronously_, which would then trigger upstream promises to also trigger synchronously (since they were all already marked resolved). 

By adding a nexttick, it allows each promise's observers to run nexttick, breaking the stack chain.